### PR TITLE
Remove anomaly rates chart.

### PR DESCRIPTION
**Pull request by vkalintiris**
##### Summary

Anomaly bits will have to queried directly from each individual chart.

##### Test Plan

N/A. Built & run locally.

### CODE REVIEW

File: database/rrd.h [hunk 1]
```diff
@@ -491,21 +491,18 @@ typedef enum rrdset_flags {
...
                                                      // least rrdset_free_obsolete_time seconds ago.
     RRDSET_FLAG_ARCHIVED                = (1 << 15),
-//    RRDSET_FLAG_ACLK                    = (1 << 16), // not used anymore
+    //                                  = (1 << 16),  reserved (not used anymore)
     RRDSET_FLAG_PENDING_FOREACH_ALARMS  = (1 << 17), // contains dims with uninitialized foreach alarms
     RRDSET_FLAG_ANOMALY_DETECTION       = (1 << 18), // flag to identify anomaly detection charts.
     RRDSET_FLAG_INDEXED_ID              = (1 << 19), // the rrdset is indexed by its id
     RRDSET_FLAG_INDEXED_NAME            = (1 << 20), // the rrdset is indexed by its name
-
-    RRDSET_FLAG_ANOMALY_RATE_CHART      = (1 << 21), // the rrdset is for storing anomaly rates for all dimensions
```
Comment (vkalintiris) [inline database/rrd.h:500]: @stelfrag I think it's safe to simply remove this. What do you think?

**Comment by vkalintiris** [comment]
@andrewm4894 keep in mind that the frontend will have to change the way they are querying anomaly rates. They'll have to get this information from each individual chart.

**Review by thiagoftsm — APPROVED** [review]
Everything is working as expected locally.

**Review by andrewm4894 — APPROVED** [review]

**Comment by andrewm4894** [comment]
> @andrewm4894 keep in mind that the frontend will have to change the way they are querying anomaly rates. They'll have to get this information from each individual chart.

??

eventually the FE will need to use the new weights endpoint instead of the `anomaly_detection.anomaly_rates` special chart. I was assuming this PR was not touching the `anomaly_detection.anomaly_rates` chart as for now that is still needed by FE. 

I just tohught this PR was to simplifiy the anomaly_detection logic.

File: database/rrd.h (hunk 1 above)
Comment (andrewm4894) [reply database/rrd.h:500]: https://github.com/netdata/netdata/pull/13763#issuecomment-1268581781

i think we should leave this in for now and then remove later when FE ready.

Comment (andrewm4894) [reply database/rrd.h:500]: fyi - i had them as seperate tasks in this plan: https://github.com/netdata/product/issues/2986

Comment (andrewm4894) [reply database/rrd.h:500]: @vkalintiris fyi

**Review by andrewm4894 — CHANGES_REQUESTED** [review]
i think we still want the `anomaly_detection.anomaly_rates` chart for now. to be removed a little later.

**Comment by andrewm4894** [comment]
cross linking to some related AA FE work we might need to do if we do decide to remove the `anomaly_detection.anoamly_rates` charts as part of this.

@vkalintiris makes a good point that it makes more sense to do this now so that we can have proper performance baselines when we move onto using multiple models at inference time and saveing and storing models in DB. So we would just need to co-ordinate the agent and FE work. 

https://github.com/netdata/product/issues/3021#issuecomment-1268617298

**Comment by andrewm4894** [comment]
oh i just realized this is a different seperate PR than this one: https://github.com/netdata/netdata/pull/13657

my comments above here assumed otherwise, apologies for the confusion.

**Comment by thiagoftsm** [comment]
Hello @vkalintiris ,

Please, rebase your PR for me and @andrewm4894 review it again.

Thanks!

**Comment by andrewm4894** [comment]
this will be a blocker to merging this: https://github.com/netdata/product/issues/3021#issuecomment-1268647105

**Comment by andrewm4894** [comment]
@vkalintiris just wondering/thinking out loud...

what if we had `/api/v1/data?chart=anomaly_detection.anomaly_rates` on agent just use `/api/v1/weights? Method=anomaly-rate` 

this would give us some backwards compatibility on agent for now and unblock us. 

no idea if might be just too messy to sort of intercept `data` calls for the `anomaly_detection.anomaly_rates` chart only and mock up a data response on the fly based on the `weights` endpoint. 

thoughts?

could solve a lot of problems but not sure how feasible might or might not be

**Comment by andrewm4894** [comment]
@vkalintiris hey - update here - @AlexNti is working on updating the AA FEC to use `/weights` so once that is all ready the work here will be unblocked. 

I'll ping in here when that is - hopefully middle next week maybe - and we can try get going on this stuff again. 

Go team!

**Comment by vkalintiris** [comment]
> What is the current status of this PR?

Ready to merge from the agent's side (after review). @andrewm4894 I recall there was a blocker from other teams when I opened this, what's the status now?

**Comment by andrewm4894** [comment]
How'd y, we just waiting on last bit of testing of the FE change to use /weights on AA endpoint and then this would be unblocked. 

I think makes sense to leave this until after next stable release either way. 

@novykh was just looking at this one https://github.com/netdata/cloud-frontend/issues/3872

Hopefully FE should be good to go soon and then can get back to this one and most importantly the next round of ml improvements in agent.

**Comment by andrewm4894** [comment]
@vkalintiris can we rebase this or resolve those conflicts? 

(just doing last few bits of testing so expecting to be clear to merge this in next day or two)

**Comment by andrewm4894** [comment]
@vkalintiris @thiagoftsm update on this is that https://staging.netdata.cloud/ is using /weights on AA tab and looking good so i expect will be deployed to production today and then we should be ok to merge this PR if we want to have it in the release.

**Comment by vkalintiris** [comment]
@andrewm4894 thanks, I'll rebase the PR then.

**Review by ilyam8 — APPROVED** [review]

**Comment by andrewm4894** [comment]
I have created some nodes in my testing room here: https://testing.netdata.cloud/spaces/testing-ml-demo/rooms/all-nodes/overview

Will compare `testing-devml-ml-enabled` and `testing-devml-ml-enabled-weightsonly` (built from this branch) to make sure expected behaviour is same and measure impact of memory savings. 

Will take a few hours to train fully, will update here with results when they all in a steady state.

**Review by MrZammler — APPROVED** [review]

**Review by andrewm4894 — APPROVED** [review]
approving as all looks good. I'm not sure I'm seeing any major memory savings from this (perhaps i'd need to let it run much longer - will leave it and see) but we still need to do it to simplify code etc.

**Comment by andrewm4894** [comment]
I do see better cpu profile on prediction in the diagnostic charts: 

image

**Review by thiagoftsm — APPROVED** [review]
I also tested locally in the past hour, and everything worked as expected. LGTM!

**Comment by ktsaou** [comment]
This PR improves Netdata in the following ways:

1. dbengine collectors memory: Before this PR each dimension was reserving 2x 4KiB, after this PR each dimension reserves 1x 4KiB. The problem was magnified with dbengine tiering. With 3 tiers, before this PR 6x 4KiB, after this PR 3x 4KiB. This memory is not immediately visible at Netdata RSS because it is taken from dbengine cache. So, dbengine cache is way less efficient before this PR, reserving double the memory for collectors.

2. internal structures memory: This is not huge, but it turns to be significant on busy parents, as each dimension before this PR needed the memory of 2 dimensions. So, at 200k dimensions, before this PR we needed 2x the structures.

3. dbengine disk footprint: Before this PR about 1/15th of the disk footprint is duplicate, increasing the bytes per point above 1. After this PR, bytes per point is a bit less than 1.

4. dbengine retention: Before this PR the retention was influenced by the additional dimensions maintained.

5. dbengine index memory: Before this PR, and for very big databases, dbengine needed way more memory on very busy parents, as the page descriptors were a lot more.

Generally, the effects of this PR are clearly visible on very busy parents.

**Review by ktsaou — APPROVED** [review]

**Pull request merged**